### ISSUE TITLE
Improve to 10/10 on Exodus

### ISSUE THREAD

**Issue by kelson42** [kiwix/kiwix-android#2859]
Our assessment on Exodus is 9/10
https://reports.exodus-privacy.eu.org/fr/reports/org.kiwix.kiwixmobile/latest/

I would like to see if we can improve to 10/10

**Comment by gouri-panda** [comment]
`ACCESS_COARSE_LOCATION` for hosting the zim files and `ACCESS_FINE_LOCATION` for zim files transfer.

**Comment by kelson42** [comment]
@gouri-panda thank you for your response. If I refer to https://developer.android.com/reference/android/Manifest.permission, I have:
* `ACCESS_COARSE_LOCATION` "Allows an app to access approximate location". I don't understand how this could be necessary for "hosting the zim files" (which I guess you mean launching a HTTP daemon and the WIFI in Hotspot mode). Would you be able to explain please to give more details?
* `ACCESS_FINE_LOCATION` "Allows an app to access precise location." I don't understand how this should be necessary for anythin... and Kiwix has not feature to transfer ZIM files (to whom? how?)

**Comment by gouri-panda** [comment]
@kelson42 We use WI-FI for hosting zim files and sharing zim files to nearby devices. A successful WI-Fi discovery peer needs these permissions to scan for available WI-Fi peers for the purpose of establishing a connection[^1].

But starting from android 13 we can make this go away by adding this permission to our manifest.  `NEARBY_WIFI_DEVICES` allows apps to manage device's connections to nearby access points over  WIFI. By adding the `neverForLocation` value we ensure that we'll never derive location information from WIFi APIs.
```
 <uses-permission android:name="android.permission.NEARBY_WIFI_DEVICES"
                     android:usesPermissionFlags="neverForLocation" />
```
But below android 13 we need these permissions.
[^1]:  https://developer.android.com/guide/topics/connectivity/wifi-scan#wifi-scan-restrictions

**Comment by kelson42** [comment]
> A successful WI-Fi discovery peer needs these permissions to scan available WI-Fi peers for the purpose of establishing a connection[](#user-content-fn-1-efe3b87118b2f4f34fc8e3200d3e9009)

Why we need to "scan" for devices? The feature is only about creating a WiFi hotspot and other users access it normally via any Web browsers. I still don't understand why we need this.

**Comment by kelson42** [comment]
@MohitMaliFtechiz Maybe you have an explanation to my last question?

**Comment by MohitMaliFtechiz** [comment]
hi @kelson42 ,

We can only get rid of these permission once we target android 13  and above  below we do requires this permission https://developer.android.com/about/versions/13/features/nearby-wifi-devices-permission. We require this permission for Peer to Peer connection https://developer.android.com/guide/topics/connectivity/wifip2p .

**Comment by kelson42** [comment]
@MohitMaliFtechiz Ok so we should change for abdroid13+. There is no way to require only this permission when this specific feature is used? To avoid a bad note?

**Comment by MohitMaliFtechiz** [comment]
> @MohitMaliFtechiz Ok so we should change for android13+. There is no way to require only this permission when this specific feature is used? To avoid a bad note?

yes currently i have checked we are requesting this permission on runtime when specific feature is used.

**Comment by kelson42** [comment]
@MohitMaliFtechiz OK. Can you adapt for 13+?

**Comment by kelson42** [comment]
@MohitMaliFtechiz Anything stopping us to move forward here?

**Comment by kelson42** [comment]
Will recheck the not once 3.8 is published on playstore.

**Comment by MohitMaliFtechiz** [comment]
This ticket is not fully fixed, and cannot be fully fixed. We need read/write permission for Android 13 and below for getting/reading the ZIM files from storage. The next ACCESS_FINE_LOCATION is necessary for Android 13 and below. Above it, we don't have this issue because we are using the `NEARBY_WIFI_DEVICES` permission.

@kelson42, I think we can move this one to next milestone.

**Comment by kelson42** [comment]
@MohitMaliFtechiz AFAIK all permissions needed listed by exodus are not listed clearly in the README. We should just do so to close thise issue IMHO.

**Comment by MohitMaliFtechiz** [comment]
@kelson42 There are mainly 3 permissions that exodus is giving an error. All those permissions are listed in the [README file](https://github.com/kiwix/kiwix-android?tab=readme-ov-file#android-permissions-needed). Apart from the listed permission other permissions are not runtime permissions, and we are not asking the user. They just need to define in the manifest to use the related features.

Like:
* INSTALL_SHORTCUT
* DYNAMIC_RECEIVER_NOT_EXPORTED_PERMISSION
* INTERNET
* FOREGROUND_SERVICE

<img width="856" height="918" alt="Image" src="https://github.com/user-attachments/assets/edb84aa6-d330-4198-9917-63d0d19ab46b" />

**Comment by kelson42** [comment]
OK  so I guess there nothing more we can do here. Closing.
